docs/undo.md

```diff
@@ -0,0 +1,280 @@
+# What should an undo system be like for TruFont?
+
+## Introduction to Undo Systems
+
+An undo/redo stack is often implemented with as an “undo manager” class.
+In the case of an outline editor, a lot of data will simply be copied, as
+it’s not always possible to simply “perform the reverse operation”.
+Transformations are a good example of that. But also point behavior: when
+a handle is implicitly being moved because it has a “smooth” connection
```

Comment:
There's no implicit movement. Every movement is performed by an algorithm. So every movement can be recorded. YOu don't even have to round in your data class, you can round in your display pipeline.

Reply:
True, I'll reformulate.

But what about transformations? If one rotates an entire glyph, would one record the changes to all points? What I was trying to say there is that recording the inverse transformation is problematic, as it will cause rounding errors in floating point numbers.

Reply:
I see, I thought you meant rounding as in the font editor rounds values to the grid. My bad.

I agree that doing floating point arithmetic in undo isn't great - that's why I rather record oldValue and newValue.

> If one rotates an entire glyph, would one record the changes to all points?

I'd do that. Also keeps the set of possible operations simpler